VCRURLConnection/VCRCassette.m

```diff
@@ -67,14 +67,30 @@ - (id)initWithData:(NSData *)data {
 - (void)addRecording:(VCRRecording *)recording {
     if (recording.URI) {
         VCRRequestKey *key = [VCRRequestKey keyForObject:recording];
-        [self.responseDictionary setObject:recording forKey:key];
+        NSMutableArray * recordings = [self.responseDictionary objectForKey:key];
+
+        if (recordings == NULL) {
+            recordings = [[NSMutableArray alloc] init];
+            [self.responseDictionary setObject:recordings forKey:key];
+        }
+
+        [recordings addObject:recording];
     } else {
         [self.regexRecordings addObject:recording];
     }
 }
 
-- (VCRRecording *)recordingForRequestKey:(VCRRequestKey *)key {
-    __block VCRRecording *recording = [self.responseDictionary objectForKey:key];
+- (VCRRecording *)recordingForRequestKey:(VCRRequestKey *)key replaying:(BOOL)replaying {
+    __block VCRRecording *recording;
+    __block NSMutableArray *recordings = [self.responseDictionary objectForKey:key];
```

Comment:
I don't see `recordings` getting mutated in a block, so this won't need the `__block` declaration.